src/multiconfigwidget.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MULTICONFIGWIDGET_H
+#define MULTICONFIGWIDGET_H
+
+#include <KConfigGroup>
+#include <functional>
+#include <QComboBox>
+#include <QPushButton>
+#include <QWidget>
+
+class MultiConfigWidget : public QWidget
+{
+    Q_OBJECT
+public:
+    explicit MultiConfigWidget(QWidget* parent = nullptr);
+
+    void setSaveFunction(const std::function<void(KConfigGroup)>& saveFunction);
```

Comment:
shouldn't both be set through the ctor already - I mean having this widget without either one leads to useless/broken behavior, no?

i.e. I think as-is this is bad API design. make both args of the ctor, _and_ assert the validity of both `std::function`'s, then remove the other runtime checks that check if either was set or not